scala/scala-impl/src/org/jetbrains/plugins/scala/lang/formatting/scalafmt/ScalafmtDynamicConfigManager.scala

```diff
@@ -224,10 +224,35 @@ object ScalafmtDynamicConfigManager {
   def isIncludedInProject(file: PsiFile): Boolean =
     instanceIn(file.getProject).isIncludedInProject(file)
 
-  def readVersion(configFile: VirtualFile): Either[Throwable, Option[String]] = {
+  private def parseHoconFile(project: Project, configFile: VirtualFile): Config = {
+    val fileDocumentManager = FileDocumentManager.getInstance
+    val emptyObj = ConfigValueFactory.fromMap(Map.empty[String, Any].asJava)
+    lazy val parseOpts: ConfigParseOptions = ConfigParseOptions.defaults
+      .setIncluder(new ConfigIncluder {
+        var fallbackOpt: Option[ConfigIncluder] = None
+        override def withFallback(fallback: ConfigIncluder): ConfigIncluder = {
+          fallbackOpt = Some(fallback)
+          this
+        }
+        override def include(context: ConfigIncludeContext,
+                             what: String): ConfigObject =
+          resolveScalafmtConfigFile(project, what)
+            .flatMap(file => fileDocumentManager.getDocument(file).toOption)
+            .map(_.getText)
+            .map(ConfigFactory.parseString(_, parseOpts))
```

Comment:
We could avoid this hack with `lazy val` and cyclic dependencies by extracting options from context. 
Have you considered using these methods:
`ConfigIncludeContext.parseOptions`
`ConfigParseable.options()` ?